### PR TITLE
New doc demonstrating layouts

### PR DESCRIPTION
**Pull request by bpaduch**
I finally got some time for this - a page that visually demonstrates the layouts in AMP.
Demo page on staging at: https://ampproject-custom.firebaseapp.com/docs/design/amp-html-layout/layouts_demonstrated

### CODE REVIEW

File: assets/sass/_content-post.scss [hunk 1]
```diff
@@ -83,6 +83,62 @@
       border: none;
     }
   }
+
+  /* styles for layouts demonstrated doc */
+  .main-layouts {
```
Comment (pbakaus) [inline assets/sass/_content-post.scss]: can we move this into another custom file? Loading this on dozens of pages might be excessive.

Comment (bpaduch) [reply assets/sass/_content-post.scss]: Ah, good point.  I'll work on that.

File: content/docs/design/amp-html-layout/layouts_demonstrated.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+              </amp-img>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <!-- flex-item -->
+      <div class="layout-item">
+        <h3>flex-item</h3> 
+        <div class="description">With a <code>fixed-height</code> layout, the element and other elements in its parent take the parent container's remaining space when the parent is a flexible container (i.e., display: <code>flex</code>).</div>
```
Comment (sebastianbenz) [inline content/docs/design/amp-html-layout/layouts_demonstrated.md]: s/fixed-height/flex-item

I don't quite understand the flex-item behaviour from the diagram and the explanation. I always thought it just makes the element behave as defined by the flex container (which usually means resizing all elements together).

Comment (bpaduch) [reply content/docs/design/amp-html-layout/layouts_demonstrated.md]: Thanks for the comments.

Based on the AMP's team's own [sample](https://github.com/ampproject/amphtml/blob/master/examples/layout-flex-item.amp.html), it doesn't look like the child items resize to "fit" the container.  Am I misunderstanding?

**Comment by bpaduch** [comment]
@pbakaus - I moved the styles into a custom file, but the styles still need to be pulled in via the docs stylesheet.  Is there a way to include only that custom .scss file for one page without having to make a separate view file (overboard imho).

**Comment by bpaduch** [comment]
@aghassemi, @jpettitt  - would be good to have your eyes on this so I am interpreting layouts correctly.

File: content/docs/design/amp-html-layout/layouts_demonstrated.md (hunk 2 above)
Comment (sebastianbenz) [reply content/docs/design/amp-html-layout/layouts_demonstrated.md]: You're right, it's just what I mostly used flex layout for. 

Suggestion: "With a <code>flex-item</code> layout, the element will take the available space based on the flexbox model."

via https://github.com/ampproject/amphtml/issues/2778

Comment (bpaduch) [reply content/docs/design/amp-html-layout/layouts_demonstrated.md]: Good wording! I've rewritten.

**Pull request merged**